app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java

```diff
@@ -0,0 +1,211 @@
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Toast;
+
+import com.mikepenz.fastadapter.FastAdapter;
+import com.mikepenz.fastadapter.IAdapter;
+import com.mikepenz.fastadapter.adapters.FastItemAdapter;
+import com.mikepenz.fastadapter.app.items.SampleItem;
+import com.mikepenz.materialize.MaterializeBuilder;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+/**
+ * TODO: Describe Class
+ *
+ * @author Ruben Gees
+ */
+public class SortActivity extends AppCompatActivity {
+
+    private static final String[] ALPHABET = new String[]{"A", "B", "C", "D", "E", "F", "G", "H",
+            "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y",
+            "Z"};
+
+    @Bind(R.id.toolbar)
+    Toolbar toolbar;
+    @Bind(R.id.rv)
+    RecyclerView recyclerView;
+
+    //save our FastAdapter
+    private FastItemAdapter<SampleItem> fastItemAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        findViewById(android.R.id.content).setSystemUiVisibility(findViewById(android.R.id.content)
+                .getSystemUiVisibility() | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sort);
+
+        ButterKnife.bind(this);
+
+        // Handle Toolbar
+        setSupportActionBar(toolbar);
+
+        //style our ui
+        new MaterializeBuilder().withActivity(this).build();
+
+        //create our FastAdapter which will manage everything
+        fastItemAdapter = new FastItemAdapter<>();
+
+        //configure our fastAdapter
+        fastItemAdapter.withOnClickListener(new FastAdapter.OnClickListener<SampleItem>() {
+            @Override
+            public boolean onClick(View v, IAdapter<SampleItem> adapter,
+                                   SampleItem item, int position) {
+                Toast.makeText(v.getContext(), (item).name.getText(v.getContext()),
+                        Toast.LENGTH_LONG).show();
+                return false;
+            }
+        });
+
+        //get our recyclerView and do basic setup
+        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.rv);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(fastItemAdapter);
+
+        if (savedInstanceState != null) {
+            //noinspection unchecked
+            fastItemAdapter.getItemAdapter().withComparator((Comparator<SampleItem>)
+                    savedInstanceState.getSerializable("comparator"));
+        }
+
+        fastItemAdapter.setNewList(generateUnsortedList());
```

Comment:
Why does `sett()` not work, whereas `setNewList()` does work if there is already a Comparator from a previous instance state?